### PR TITLE
Allure test filter

### PR DESCRIPTION
**Pull request by Litemn**

### CODE REVIEW

**Comment by Malinskiy** [comment]
Hey @Litemn . Thanks for this! I believe I've done the initial implementation for this months ago but didn't receive any feedback. Here is the branch https://github.com/MarathonLabs/marathon/tree/feature/allure-test-filter and the actual commit https://github.com/MarathonLabs/marathon/commit/365e052b360886dd779a5c71fc0d644f3f0b6637.

File: core/src/main/kotlin/com/malinskiy/marathon/report/allure/AllureReporter.kt [hunk 1]
```diff
@@ -64,7 +64,7 @@ class AllureReporter(val configuration: Configuration, private val outputDirecto
 
     private fun createTestResult(uuid: String, device: DeviceInfo, testResult: TestResult): io.qameta.allure.model.TestResult {
         val test = testResult.test
-        val fullName = test.toSimpleSafeTestName()
```
Comment (Malinskiy) [inline core/src/main/kotlin/com/malinskiy/marathon/report/allure/AllureReporter.kt]: This change is not related to the test filter, please remove :)

**Comment by Malinskiy** [comment]
Can you also please add the relevant documentation changes from the old branch?

File: configuration/src/main/kotlin/com/malinskiy/marathon/config/FilteringConfiguration.kt [hunk 2]
```diff
@@ -300,4 +301,17 @@ sealed class TestFilterConfiguration {
 
         override fun hashCode(): Int = filters.hashCode() + op.hashCode()
     }
+
+    class AllureFilterConfiguration : TestFilterConfiguration() {
```
Comment (Malinskiy) [inline configuration/src/main/kotlin/com/malinskiy/marathon/config/FilteringConfiguration.kt]: Maybe just create an `object` here since there are no parameters?

File: core/src/main/kotlin/com/malinskiy/marathon/report/allure/AllureReporter.kt (hunk 1 above)
Comment (Litemn) [reply core/src/main/kotlin/com/malinskiy/marathon/report/allure/AllureReporter.kt]: Thanks!

File: configuration/src/main/kotlin/com/malinskiy/marathon/config/FilteringConfiguration.kt (hunk 2 above)
Comment (Litemn) [reply configuration/src/main/kotlin/com/malinskiy/marathon/config/FilteringConfiguration.kt]: Done

**Review by Malinskiy — APPROVED** [review]

**Review by Malinskiy — APPROVED** [review]

**Review by Malinskiy — APPROVED** [review]
